BookingsApi/BookingsApi/Controllers/HearingsController.cs

```diff
@@ -115,6 +115,36 @@ public async Task<IActionResult> GetHearingsByUsername([FromQuery]string usernam
             return Ok(response);
         }
 
+        /// <summary>
+        /// Get list of all hearings in a group
+        /// </summary>
+        /// <param name="groupId">the group id of the single day or multi day hearing</param>
+        /// <returns>Hearing details</returns>
+        [HttpGet(Name = "GetHearingsByGroupId")]
+        [OpenApiOperation("GetHearingsByGroupId")]
+        [ProducesResponseType(typeof(List<HearingDetailsResponse>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetHearingsByGroupId([FromQuery]Guid groupId)
+        {
+            try
+            {
+                var query = new GetHearingsByGroupIdQuery(groupId);
+                var hearings = await _queryHandler.Handle<GetHearingsByGroupIdQuery, List<VideoHearing>>(query);
+
+                var hearingMapper = new HearingToDetailResponseMapper();
+                var response = hearings.Select(hearingMapper.MapHearingToDetailedResponse).ToList();
+                
+                return Ok(response);
+            }
+            catch (Exception e)
```

Comment:
What exceptions do you anticipate here? The current `GetHearingDetailsById` doesn't have a try-catch.